### PR TITLE
Add $defaultTheme property if missing on BrowserTestBase

### PR DESCRIPTION
**Pull request by mglaman**
## Description
Add `protected $defaultTheme = 'stark';` to functional tests if missing.

This could cause test failures because they assume on markup for Classy, which
the theme should be changed to `starterkit` as Classy is going away in Drupal 10.

## Drupal.org issue
https://www.drupal.org/project/rector/issues/3299835

### CODE REVIEW

File: config/drupal-8/drupal-8.8-deprecations.php [hunk 1]
```diff
@@ -33,4 +34,6 @@
     $rectorConfig->rule(FileScanDirectoryRector::class);
     $rectorConfig->rule(FileDirectoryTempRector::class);
     $rectorConfig->rule(FileUriTargetRector::class);
+
+    $rectorConfig->rule(FunctionalTestDefaultThemePropertyRector::class);
```
Comment (mglaman) [inline config/drupal-8/drupal-8.8-deprecations.php]: Well. This should have failed in the fixtures: `fixtures/d8/rector_examples/test/src/Functional/BrowserTestBaseGetMock.php`

```
<?php

namespace Drupal\Tests\rector_examples\Functional;

use Drupal\Core\Entity\EntityTypeManagerInterface;
use Drupal\Tests\BrowserTestBase;

class BrowserTestBaseGetMock extends BrowserTestBase {

  /**
   * A simple example using the class property.
   */
  public function simple_example() {
      $this->entityTypeManager = $this->getMock(EntityTypeManagerInterface::class);
  }

  /**
   * A simple example using a string directly.
   */
  public function class_name_as_string() {
    $this->entityTypeManager = $this->getMock('Drupal\Core\Entity\EntityTypeManagerInterface');
  }
}
```

**Comment by mglaman** [comment]
It's running, but not making changes:

```
2022-08-10T16:07:45.9590491Z [file] web/modules/custom/rector_examples/test/src/Functional/BrowserTestBaseGetMock.php
2022-08-10T16:07:45.9590707Z [rule] DrupalRector\Rector\Class_\FunctionalTestDefaultThemePropertyRector
```

And here:

```
2022-08-10T16:07:44.9331621Z [file] web/modules/custom/rector_examples/tests/src/Functional/PublicStaticModulesTest.php
2022-08-10T16:07:44.9332120Z [file] web/modules/custom/rector_examples/tests/src/Functional/AssertFieldByIdTest.php
2022-08-10T16:07:44.9332384Z [rule] DrupalRector\Rector\Class_\FunctionalTestDefaultThemePropertyRector
2022-08-10T16:07:44.9332594Z [rule] DrupalRector\Rector\Deprecation\AssertElementPresentRector
```

File: src/Rector/Class_/FunctionalTestDefaultThemePropertyRector.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        if ($type->isSuperTypeOf(new ObjectType(BrowserTestBase::class))->no()) {
+            return null;
+        }
+
+        // Is processed by \Rector\PostRector\Rector\PropertyAddingPostRector
+        // Sets as `private`, which we need `protected` and default value.
+        $propertyBuilder = new PropertyBuilder('defaultTheme');
+        $propertyBuilder->makeProtected();
+        $propertyBuilder->setDefault('stark');
```
Comment (mglaman) [inline src/Rector/Class_/FunctionalTestDefaultThemePropertyRector.php]: We need to check if any classes the test extends has the property with a value.

File: src/Rector/Class_/FunctionalTestDefaultThemePropertyRector.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        $propertyBuilder->setDocComment("/**\n * {@inheritdoc}\n */");
+        $property = $propertyBuilder->getNode();
+        $this->phpDocInfoFactory->createFromNode($property);
+        $node->stmts = array_merge([$property], $node->stmts);
```
Comment (mglaman) [inline src/Rector/Class_/FunctionalTestDefaultThemePropertyRector.php]:
```
"web/modules/custom/rector_examples/tests/src/Functional/AssertLinkTest.php"
         file, due to:                                                          
         "System error: "Class 'Symplify\Astral\ValueObject\NodeBuilder\PropertyBuilder'
         not found"  
```

**Comment by agentrickard** [comment]
@mglaman This one has a merge conflict I am uncertain about

**Comment by bbrala** [comment]
This was one of the top issues last bot run. So thought I'd update and fix it. All seems fine now.

**Review by mglaman — COMMENTED** [review]
I approve of the changes 👍 but it's my own PR so, this is my +1 for someone to approve and merge

**Review by bbrala — APPROVED** [review]
As per matt, approved

**Pull request merged**